scalecube-configuration/src/main/java/io/scalecube/configuration/ConfigurationServiceImpl.java

```diff
@@ -1,133 +1,100 @@
-    this.authorizationService = AuthorizationService.builder().build();
+  public ConfigurationServiceImpl(ConfigurationRepository repository, AccessControl accessContorl) {
+    this.repository = repository;
+    this.accessControl = accessContorl;
   }
 
   @Override
   public Mono<Acknowledgment> createRepository(CreateRepositoryRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("createRepository: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::createRepository))
-        .flatMap(operation -> Mono
-            .from(operation.execute(request, context(OperationType.CreateRepoitory))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(result -> logger.debug("createRepository: exit: request: {}", request))
-        .doOnError(th -> logger.error("createRepository: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_CREATE_REPO)
+        .flatMap(
+            p -> this.repository.createRepository(new Repository(p.tenant(), request.repository())))
+        .map(b -> ACK);
   }
 
   @Override
   public Mono<FetchResponse> fetch(FetchRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("fetch: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::fetch))
-        .flatMap(operation -> Mono.from(operation.execute(request, context(OperationType.Read))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnSuccess(
-            response -> logger.debug("fetch: exit: request: {}, response: {}", request, response))
-        .doOnError(th -> logger.error("fetch: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_FETCH)
+        .map(p -> this.repository.fetch(p.tenant(), request.repository(), request.key()))
+        .flatMap(
+            document ->
+                document
+                    .map(Document::value)
+                    .map(value -> new FetchResponse(request.key(), value)));
   }
 
   @Override
   public Flux<FetchResponse> entries(FetchRequest request) {
-    return Mono.fromRunnable(() -> logger.debug("entries: enter: request: {}", request))
-        .then(Mono.fromCallable(ServiceOperationFactory::fetchAll))
-        .flatMapMany(
-            operation -> Flux.from(operation.execute(request, context(OperationType.List))))
-        .subscribeOn(Schedulers.parallel())
-        .doOnComplete(() -> logger.debug("entries: exit: request: {}", request))
-        .doOnError(th -> logger.error("entries: request: {}, error: {}", request, th));
+    verifyRequest(request);
+    return accessControl
+        .check(request.token().toString(), ConfigurationService.CONFIG_ENTRIES)
+        .map(p -> this.repository.fetchAll(p.tenant(), request.repository()))
```

Comment:
also here, if the `check` is heavy I guess better to use `flatMap` as in previous cases